### PR TITLE
feat: always separate lidar preprocessing from pointcloud_container

### PR DESCRIPTION
**Pull request by kminoda**
## Description
See https://github.com/autowarefoundation/autoware.universe/pull/6091

## Related links
See https://github.com/autowarefoundation/autoware.universe/pull/6091

## Tests performed
See https://github.com/autowarefoundation/autoware.universe/pull/6091

## Notes for reviewers
See https://github.com/autowarefoundation/autoware.universe/pull/6091

### Note for X2 team (TIER IV)
The modification in tier4_localization_component.launch.xml should be incorporated into autoware_launch.x2 carefully. Specifically, please change the target container name from `/sensing/lidar/top/...` to the one you use for localization.

## Interface changes
See https://github.com/autowarefoundation/autoware.universe/pull/6091

## Effects on system behavior
See https://github.com/autowarefoundation/autoware.universe/pull/6091

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_launch/launch/components/tier4_localization_component.launch.xml [hunk 1]
```diff
@@ -4,15 +4,16 @@
   <arg name="pose_source" default="ndt" description="select pose_estimator: ndt, yabloc, eagleye"/>
   <arg name="twist_source" default="gyro_odom" description="select twist_estimator. gyro_odom, eagleye"/>
   <arg name="input_pointcloud" default="/sensing/lidar/top/pointcloud" description="The topic will be used in the localization util module"/>
+  <arg name="lidar_container_name" default="/sensing/lidar/top/pointcloud_preprocessor/pointcloud_container" description="The topic will be used in the localization util module"/>
```
Comment (KYabuuchi) [inline autoware_launch/launch/components/tier4_localization_component.launch.xml:7]: Please update the description :pray:

File: autoware_launch/launch/components/tier4_localization_component.launch.xml [hunk 2]
```diff
@@ -4,15 +4,20 @@
...
+  <arg
+    name="lidar_container_name"
+    default="/sensing/lidar/top/pointcloud_preprocessor/pointcloud_container"
+    description="The target container to which lidar preprocessing nodes in localization be attached"
+  />
 
   <group>
     <include file="$(find-pkg-share tier4_localization_launch)/launch/localization.launch.xml">
-      <arg name="use_pointcloud_container" value="$(var use_pointcloud_container)"/>
       <arg name="pointcloud_container_name" value="$(var pointcloud_container_name)"/>
```
Comment (KYabuuchi) [inline autoware_launch/launch/components/tier4_localization_component.launch.xml:15]: `pointcloud_container_name` is no longer used in `launch/tier4_localization_launch/launch/localization.launch.xml`

```suggestion
```

image

Comment (kminoda) [reply autoware_launch/launch/components/tier4_localization_component.launch.xml]: Right, thanks. Fixed: https://github.com/autowarefoundation/autoware_launch/pull/796/commits/04ee9e57beab9d9906affe11e6db4509b4baa062

**Review by miursh — APPROVED** [review]
LGTM

**Pull request merged**